src/analyses/base.ml

```diff
@@ -1407,12 +1407,14 @@ struct
         let m = ID.meet a b in
         (match op, ID.to_bool c with
         | Eq, Some true
-        | Ne, Some false -> both m (* def. equal *)
+        | Ne, Some false -> both m (* def. equal: if they compare equal, both values must be from the meet *)
         | Eq, Some false
         | Ne, Some true -> (* def. unequal *)
-          (match ID.to_int m with
-          | Some i -> both (ID.of_excl_list ik [i])
-          | None -> a, b)
+          (* if they compare unequal, they can not at the same time be from the meet, but it would be unsound to restrict both to not be the meet      *)
+          (* even if there is only one element in the meet e.g. a:[0,1] b:[1,2] meet(a,b) = [1], but (a != b) does not mean that a:[0,0] and b: [2,2]  *)
```

Comment:
Why not?

Reply:
If you consider the example above, `a` could be `1` and `b` could be `2`.
Restricting `a` and `b` with `Not{1}` would lead to `bot` for `a`. Which would mean we raise `Deadcode` later when we get to `invariant` for `Lval a`.

Reply:
Ah, ok, yes.

Reply:
The example where there were problems was:

```
int x;
int two = 2;

if( x== two) {

} else {
   // Unreachable
}
```

In this case we restricted `Definite 2` with `Not{2}` leading to `bot` for `two` and then raising `Deadcode` from the `invariant` code for `LVal`.